crates/webui-tester/src/ui.rs

```diff
@@ -0,0 +1,944 @@
+    create: ElementResolver<WebElement>,
+}
+
+impl AddGenericConnectorDialog {
+    pub async fn set_name(&self, name: &str) -> WebDriverResult<()> {
+        let elem = self.form_name.resolve().await?;
+        elem.clear().await?;
+        elem.send_keys(name).await?;
+        Ok(())
+    }
+
+    pub async fn set_description(&self, description: &str) -> WebDriverResult<()> {
+        let elem = self.form_description.resolve().await?;
+        elem.clear().await?;
+        elem.send_keys(description).await?;
+        Ok(())
+    }
+
+    pub async fn set_config(&self, config: &str) -> WebDriverResult<()> {
+        let elem = self.form_config.resolve().await?;
+        elem.focus().await?;
+
+        // Remove all pre-existing text:
+        //
+        // This doesn't seem to have any effect -- the editor is a weird
+        // combination of divs:
+        elem.clear().await?;
+        // Instead we do this which works:
+        for _i in 0..120 {
+            // TODO: get length by reading current text
+            elem.send_keys("" + &Key::Delete).await?;
+        }
+
+        // Insert the new text
+        elem.send_keys(config).await?;
+
+        Ok(())
+    }
+}
+
+/// Connect a input handle to a table.
+///
+/// # Notes on why this function is so complicated
+///
+/// - The straight forward way would be to use `drag_and_drop_element` -- but it
+///   didn't work. See also https://github.com/SeleniumHQ/selenium/issues/8003 and
+///   https://github.com/w3c/webdriver/issues/1488
+/// - The "fix" in thirtyfour is `js_drag_to` -- didn't work.
+/// - A bunch of other javascript hacks with xpath / no jquery etc. --
+///   didn't work See also
+///   https://stackoverflow.com/questions/71188974/how-to-perform-drag-and-drop-on-html5-element-using-selenium-3-141-59-java
+///
+/// What did end up working is this:
+/// https://github.com/SeleniumHQ/selenium/issues/3269#issuecomment-452757044
+/// e.g., put a bunch of sleeps around the actions.
+pub async fn drag_and_drop_impl(
+    driver: &WebDriver,
+    from: &WebElement,
+    to: &WebElement,
+) -> Result<()> {
+    // TODO: the 800ms seems high but if we go lower (e.g., 200ms) it looks
+    // like the browser is still connecting things but the attachment is not
+    // saved properly in the backend, unclear yet if this is a bug in the UI
+    // or still the drag-and-drop bug in selenium
+    driver
+        .action_chain()
+        .move_to_element_center(&from)
+        .perform()
+        .await?;
+    tokio::time::sleep(Duration::from_millis(800)).await;
+    driver
+        .action_chain()
+        .click_and_hold_element(&from)
+        .perform()
+        .await?;
+    tokio::time::sleep(Duration::from_millis(800)).await;
+    driver
+        .action_chain()
+        .move_to_element_center(&to)
+        .perform()
+        .await?;
+    tokio::time::sleep(Duration::from_millis(800)).await;
+    driver
+        .action_chain()
+        .move_by_offset(-1, -1)
```

Comment:
This is also only necessary because it's a workaround due to a selenium bug. Once it's fixed this function can be just replaced by calling the regular drag_and_drop function.